File: Gui/src/main/java/net/codersoffortune/infinity/gui/InfinityToolsGui.java
package net.codersoffortune.infinity.gui;

import javafx.application.Application;
import javafx.fxml.FXMLLoader;
import javafx.scene.Scene;
import javafx.scene.layout.Pane;
import javafx.stage.Stage;
import org.apache.logging.log4j.Level;
import org.apache.logging.log4j.core.config.Configurator;
import java.io.IOException;

public class InfinityToolsGui extends Application {
    private static Stage stage = null;

    @Override
    public void start(Stage stage) throws IOException {
        Pane anchorPane = FXMLLoader.load(InfinityToolsGui.class.getResource("/Main.fxml"));
        Scene scene = new Scene(anchorPane);
        stage.setTitle("Infinity Tools Gui");
        stage.setScene(scene);
        stage.show();
        InfinityToolsGui.stage = stage;
    }

    public static void switchWindow(String name) throws IOException {
        String fileName = String.format("/%s.fxml", name);
        Pane pane = FXMLLoader.load(InfinityToolsGui.class.getResource(fileName));
        stage.getScene().setRoot(pane);
    }

    public static void main(String[] args) {
        Configurator.setRootLevel(Level.DEBUG);
        launch();
    }

}

File: Gui/src/main/kotlin/net/codersoffortune/infinity/gui/InfinityToolsGui.kt
package net.codersoffortune.infinity.gui

import javafx.application.Application
import javafx.fxml.FXMLLoader
import javafx.scene.Scene
import javafx.scene.layout.Pane
import javafx.stage.Stage
import org.apache.logging.log4j.Level
import org.apache.logging.log4j.core.config.Configurator

class InfinityToolsGui : Application() {
    override fun start(stage: Stage) {
        val pane = FXMLLoader.load<Pane>(InfinityToolsGui::class.java.getResource("/Main.fxml"))
        stage.scene = Scene(pane)
        stage.title = "Infinity Tools Gui"
        stage.show()
        Companion.stage = stage
    }

    companion object {
        private var stage: Stage? = null

        @JvmStatic
        fun main(args: Array<String>) {
            Configurator.setRootLevel(Level.DEBUG)
            launch(InfinityToolsGui::class.java)
        }

        fun switchWindow(name: String) {
            val pane = FXMLLoader.load<Pane>(InfinityToolsGui::class.java.getResource("/$name.fxml"))
            stage!!.scene.root = pane
        }
    }
}
